### DISCUSSION TITLE
IP adresss of requestor (for rate limitors)

### DISCUSSION THREAD

**Discussion by ollyde** [trpc/trpc#2596] [General]
Trying to retro fit our GraphQL rate limiter into tRPC, but one requirement is that we can access the clients IP address.

How can we access their IP address in tRPC?

**Comment by ollyde** [comment]
Figure it out, you can forward the req object which has the "ip" in there.

```
export async function authContext({
  req,
  res,
}: trpcNext.CreateNextContextOptions) {
  return {
    req,
  };
}
export type Context = inferAsyncReturnType<typeof authContext>;
```

Can then be accessed via the routes


```
export const userSelf = trpc.router<Context>().query('self', {
  async resolve({ ctx }) {
    rateLimit({ windowInMinutes: 1, requestsAllowed: 10, ip: ctx.req.ip });
  },
});
```

**Comment by ollyde** [comment]
Although the IP address looks a bit funky, with the standard GraphQL or express setups we get the localhost IP, here we get "::1" not sure if this is something to be worried about 🤪

**Reply by nitasn** [reply]
"::1" in IPv6 =  "127.0.0.1" in IPv4
